packages/shader-lab/scripts/GenDts.ts

```diff
@@ -6,7 +6,13 @@ import fs from "fs";
 function genDts(opts?: { outDir?: string }) {
   const parser = new ShaderParser();
   const productions = parser.getGAstProductions();
-  const dtsContent = generateCstDts(productions);
+  let dtsContent = generateCstDts(productions);
+
+  dtsContent = dtsContent.replace(
+    /((BlendFactor|BlendOperation|CompareFunction|StencilOperation|CullMode)\.\w+?)\?:/g,
+    '"$1"?:'
+  );
```

Comment:
something hack?

Reply:
Yeah, the type generation tool of chevrotain does not fully support the declaration of complex type properties. The code above will not run at runtime